File: Sequence Valley/Prefix Sum/Q3. Ways to Make a Fair Array/Q3_WaystoMakeaFairArray.py
class Solution(object):
    def waysToMakeFair(self, nums):
        """
        :type nums: List[int]                  # nums is a list of integers
        :rtype: int                            # function returns an integer
        """
        evens = sum(nums[0::2])                # Sum of elements at even indices
        odds = sum(nums[1::2])                 # Sum of elements at odd indices
        ans = 0                                # Count of fair indices
        toggle = 0                             # Tracks index parity

        for n in nums:                         # Loop through each element
            if toggle:                         # If index is odd
                odds -= n                     # Remove n from odd sum
                ans += (evens == odds)         # Check if sums are equal
                evens += n                    # Add n to even sum

            else:                              # If index is even
                evens -= n                    # Remove n from even sum
                ans += (evens == odds)         # Check if sums are equal
                odds += n                     # Add n to odd sum

            toggle ^= 1                        # Switch between even and odd

        return ans                             # Return number of fair indices

File: Sequence Valley/Prefix Sum/Q3. Ways to Make a Fair Array/Q3_WaystoMakeaFairArray.java
class Solution {                                // Define Solution class
    public int waysToMakeFair(int[] nums) {     // Method to count fair indices
        int esum = 0;                           // Sum of elements at even positions after removal
        int osum = 0;                           // Sum of elements at odd positions after removal
        int n = nums.length;                    // Store length of the array

        for (int i = 0; i < n; i++) {            // Loop through the array
            if (i % 2 == 0) {                    // If index is even
                osum += nums[i];                 // Add value to odd sum (will shift after removal)
            } else {                             // If index is odd
                esum += nums[i];                 // Add value to even sum (will shift after removal)
            }
        }

        int count = 0;                           // Stores number of fair indices
        int prev = 0;                            // Stores previous element

        for (int i = 0; i < n; i++) {            // Loop through each index
            if (i % 2 == 0) {                    // If current index is even
                osum = osum - nums[i] + prev;    // Update odd sum after removing nums[i]
            } else {                             // If current index is odd
                esum = esum - nums[i] + prev;    // Update even sum after removing nums[i]
            }

            if (esum == osum) {                  // Check if array becomes fair
                count++;                         // Increase fair index count
            }

            prev = nums[i];                      // Update previous element
        }

        return count;                            // Return total fair indices
    }
}
